src/vs/workbench/electron-sandbox/desktop.main.ts

```diff
@@ -241,9 +241,18 @@ export class DesktopMain extends Disposable {
 		// User Data Profiles
 		const userDataProfilesService = new UserDataProfilesNativeService(this.configuration.profiles.all, mainProcessService, environmentService);
 		serviceCollection.set(IUserDataProfilesService, userDataProfilesService);
-		const userDataProfileService = new UserDataProfileService(reviveProfile(this.configuration.profiles.current, userDataProfilesService.profilesHome.scheme), userDataProfilesService);
+		const userDataProfileService = new UserDataProfileService(this.configuration.profiles.profile ? reviveProfile(this.configuration.profiles.profile, userDataProfilesService.profilesHome.scheme) : userDataProfilesService.defaultProfile, userDataProfilesService);
 		serviceCollection.set(IUserDataProfileService, userDataProfileService);
 
+		const payload = this.resolveWorkspaceInitializationPayload(environmentService);
```

Comment:
I would not expect this code to live here but in main, shouldnt we create the profile in main and then open the window or are you worried about perf?

Reply:
I started with creating and associating the profile in the main but I realised that it won't scale with following new features to add:

- Open a workspace with a profile creating from a template file
- Open a workspace with a profile created from another profile

Supporting them in main is complicated when compared to Renderer which already handles various profile creation scenarios. Another not so nice point in main is that profile creation has to be done at two places.